src/ctapipe/utils/astro.py

```diff
@@ -4,63 +4,218 @@
+    band: string
+        wavelength band to use for the magnitude cut options are V and B. Default: 'B' (all entries)
+
+    Returns
+    -------
+    Astropy table:
+       List of all stars after cuts with same keys as the input table stars
+    """
+
+    if magnitude_cut is not None:
+        if band == "B":
+            if "Bmag" in stars.keys():
+                stars = stars[stars["Bmag"] < magnitude_cut]
+            elif "BTmag" in stars.keys():
+                stars = stars[stars["BTmag"] < magnitude_cut]
+            else:
+                raise ValueError(
+                    "The requested catalogue has no compatible magnitude for the B band"
+                )
 
-__all__ = ["get_bright_stars"]
+        if band == "V":
+            if "Vmag" in stars.keys():
+                stars = stars[stars["Vmag"] < magnitude_cut]
+            elif "VTmag" in stars.keys():
+                stars = stars[stars["VTmag"] < magnitude_cut]
+            else:
+                raise ValueError(
+                    "The requested catalogue has no compatible magnitude for the V band"
+                )
+
+    if radius is not None:
+        if pointing is None:
+            raise ValueError(
+                "Sky pointing, pointing=SkyCoord(), must be "
+                "provided if radius is given."
+            )
+        separations = stars["ra_dec"].separation(pointing)
+        stars["separation"] = separations
+        stars = stars[separations < radius]
 
+    return stars
 
-def get_bright_stars(pointing=None, radius=None, magnitude_cut=None):
+
+def get_star_catalog(catalog, min_magnitude=0.0, max_magnitude=10.0, row_limit=1000000):
+    """
+    Utility function to download a star catalog for the get_bright_stars function
+
+    Parameters
+    ----------
+    catalog: string
+        Name of the catalog to be used. Usable names are found in the Enum StarCatalogues. Default: Yale
+    min_magnitude: float
+        Minimum value for magnitude used in lookup
+    max_magnitude: float
+        Maximum value for magnitude used in lookup
+    row_limit: int
+        Maximum number of rows for the star catalog lookup
+
+    Returns
+    ----------
+    Astropy table:
+       List of all stars after cuts with catalog numbers, magnitudes,
+       and coordinates as SkyCoord objects including proper motion
     """
-    Get an astropy table of bright stars.
+    from astroquery.vizier import Vizier
 
-    This function is using the Yale bright star catalog, available through ctapipe
-    data downloads.
+    catalog_dict = StarCatalogues[catalog].value
 
-    The included Yale bright star catalog contains all 9096 stars, excluding the
-    Nova objects present in the original catalog and is complete down to magnitude
-    ~6.5, while the faintest included star has mag=7.96. :cite:p:`bright-star-catalog`
+    columns = ["RAJ2000", "DEJ2000", "pmRA", "pmDE", catalog_dict["ID_type"]]
+    if "B" in catalog_dict["band"]:
+        columns.append("Bmag")
```

Comment:
Is it an error to request columns that do not exist? If not, why this logic instead of requesting all possible columns we need for both?